tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/service/websocket/RetryClientTest.java

```diff
@@ -0,0 +1,166 @@
+            ", sending, and receiving binary frames then restart that server and repeat.)")
+    public void testBinaryFrameForRetryWithMaxCount() throws URISyntaxException, InterruptedException,
+            BallerinaTestException {
+        CountDownLatch countDownLatch = new CountDownLatch(1);
+        ByteBuffer bufferSent = ByteBuffer.wrap(new byte[]{1, 2, 3, 4, 5});
+        remoteServer = new WebSocketRemoteServer(port);
+        remoteServer.run();
+        WebSocketTestClient client = new WebSocketTestClient("ws://localhost:21031");
+        client.handshake();
+        client.setCountDownLatch(countDownLatch);
+        client.sendBinary(bufferSent);
+        countDownLatch.await(TIMEOUT_IN_SECS, TimeUnit.SECONDS);
+        Assert.assertEquals(client.getBufferReceived(), bufferSent);
+        remoteServer.stop();
+        CountDownLatch latchForRestart = new CountDownLatch(1);
+        latchForRestart.await(TIMEOUT_IN_SECS, TimeUnit.SECONDS);
+        remoteServer.run();
+        client.sendBinary(bufferSent);
+        latchForRestart.await(TIMEOUT_IN_SECS, TimeUnit.SECONDS);
+        Assert.assertNull(client.getBufferReceived());
+        client.shutDown();
+        remoteServer.stop();
+    }
+
+    @Test(description = "Tests the `countDownLatch` for the retry function using the WebSocket client (starting the" +
+            " server, sending, and receiving binary frames then restart that server and wait for six minutes.)")
+    public void testCoundownLatchForRetry() throws URISyntaxException, InterruptedException,
+            BallerinaTestException {
+        CountDownLatch countDownLatch = new CountDownLatch(1);
+        ByteBuffer bufferSent = ByteBuffer.wrap(new byte[]{1, 2, 3, 4, 5});
+        remoteServer = new WebSocketRemoteServer(port);
+        remoteServer.run();
+        WebSocketTestClient client = new WebSocketTestClient("ws://localhost:21032");
+        client.handshake();
+        client.setCountDownLatch(countDownLatch);
+        client.sendBinary(bufferSent);
+        countDownLatch.await(TIMEOUT_IN_SECS, TimeUnit.SECONDS);
+        Assert.assertEquals(client.getBufferReceived(), bufferSent);
+        remoteServer.stop();
+        CountDownLatch latchForRestart = new CountDownLatch(1);
+        latchForRestart.await(320, TimeUnit.SECONDS);
+        remoteServer.run();
+        latchForRestart.await(TIMEOUT_IN_SECS, TimeUnit.SECONDS);
+        client.sendBinary(bufferSent);
+        latchForRestart.await(TIMEOUT_IN_SECS, TimeUnit.SECONDS);
+        Assert.assertEquals(client.getBufferReceived(), bufferSent);
+        client.shutDown();
+        remoteServer.stop();
+    }
+
+    @Test(description = "Tests the function for retrying using the WebSocket client. " +
+            "When `readyOnConnect` is true, calls the `ready()` function.")
+    public void testReadyOnConnect() throws URISyntaxException, InterruptedException,
```

Comment:
Does this test case belong with the Retry test cases? Can we try to move it to a suitable place?

Reply:
This test moved to  https://github.com/ballerina-platform/ballerina-lang/pull/19723/files#diff-d96eaa3b156410bc49f451e8f95d4cf2R96